src/dtclient/kubernetes_settings_test.go

```diff
@@ -0,0 +1,389 @@
+
+		// assert
+		assert.NoError(t, err)
+		assert.NotNil(t, actual)
+		assert.True(t, actual.TotalCount == 0)
+	})
+
+	t.Run(`no settings found for because of an api error`, func(t *testing.T) {
+		// arrange
+		entities := createMonitoredEntities()
+		// it is immaterial what we put here since the http request is producing an error
+		totalCount := 999
+
+		dynatraceServer := httptest.NewServer(dynatraceServerSettingsHandler(totalCount, "", true))
+		defer dynatraceServer.Close()
+
+		skipCert := SkipCertificateValidation(true)
+		dtc, err := NewClient(dynatraceServer.URL, apiToken, paasToken, skipCert)
+		require.NoError(t, err)
+		require.NotNil(t, dtc)
+
+		// act
+		actual, err := dtc.(*dynatraceClient).GetSettingsForMonitoredEntities(entities)
+
+		// assert
+		assert.Error(t, err)
+		assert.True(t, actual.TotalCount == 0)
+	})
+}
+
+func TestDynatraceClient_CreateKubernetesSetting(t *testing.T) {
+	t.Run(`create settings for the given monitored entity id`, func(t *testing.T) {
+		// arrange
+		dynatraceServer := httptest.NewServer(dynatraceServerSettingsHandler(1, testObjectID, false))
+		defer dynatraceServer.Close()
+
+		skipCert := SkipCertificateValidation(true)
+		dtc, err := NewClient(dynatraceServer.URL, apiToken, paasToken, skipCert)
+		require.NoError(t, err)
+		require.NotNil(t, dtc)
+
+		// act
+		actual, err := dtc.(*dynatraceClient).CreateKubernetesSetting(testName, testUID, testScope)
+
+		// assert
+		assert.NotNil(t, actual)
+		assert.NoError(t, err)
+		assert.Len(t, actual, len(testObjectID))
+		assert.EqualValues(t, testObjectID, actual)
+	})
+
+	t.Run(`don't create settings for the given monitored entity id because no kube-system uuid is provided`, func(t *testing.T) {
+		// arrange
+		dynatraceServer := httptest.NewServer(dynatraceServerSettingsHandler(1, testObjectID, false))
+		defer dynatraceServer.Close()
+
+		skipCert := SkipCertificateValidation(true)
+		dtc, err := NewClient(dynatraceServer.URL, apiToken, paasToken, skipCert)
+		require.NoError(t, err)
+		require.NotNil(t, dtc)
+
+		// act
+		actual, err := dtc.(*dynatraceClient).CreateKubernetesSetting(testName, "", testScope)
+
+		// assert
+		assert.Error(t, err)
+		assert.Len(t, actual, 0)
+	})
+
+	t.Run(`don't create settings for the given monitored entity id because of api error`, func(t *testing.T) {
+		// arrange
+		dynatraceServer := httptest.NewServer(dynatraceServerSettingsHandler(1, testObjectID, true))
+		defer dynatraceServer.Close()
+
+		skipCert := SkipCertificateValidation(true)
+		dtc, err := NewClient(dynatraceServer.URL, apiToken, paasToken, skipCert)
+		require.NoError(t, err)
+		require.NotNil(t, dtc)
+
+		// act
+		actual, err := dtc.(*dynatraceClient).CreateKubernetesSetting(testName, testUID, testScope)
+
+		// assert
+		assert.Error(t, err)
+		assert.Len(t, actual, 0)
+	})
+}
+
+func handleEntitiesRequest(request *http.Request, writer http.ResponseWriter, entities []MonitoredEntity) {
```

Comment:
these are mocks right ?
This name could be confused with actual (not test) code, so naming like `mockHandleEntitiesRequest ` or something similar

Reply:
see comment below, I was not sure, I will check again if it is a mock or a fake